ledger_secure_sdk_sys/build.rs

```diff
@@ -71,48 +71,212 @@ const CCID_FILES: [&str; 9] = [
+
+fn retrieve_sdk_git_info(bolos_sdk: &Path) -> (String, String) {
+    let c_sdk_hash = match Command::new("git")
+        .arg("-C")
+        .arg(bolos_sdk)
+        .arg("describe")
+        .arg("--always")
+        .arg("--dirty")
+        .arg("--exclude")
+        .arg("*")
+        .arg("--abbrev=40")
+        .output()
+        .ok()
+    {
+        Some(output) => {
+            if output.stdout.is_empty() {
+                "None".to_string()
+            } else {
+                String::from_utf8(output.stdout).unwrap_or("None".to_string())
+            }
+        }
+        None => "None".to_string(),
+    };
+
+    let c_sdk_version = match Command::new("git")
+        .arg("-C")
+        .arg(bolos_sdk)
+        .arg("describe")
+        .arg("--tags")
+        .arg("--exact-match")
+        .arg("--match")
+        .arg("v[0-9]*")
+        .arg("--dirty")
+        .output()
+        .ok()
+    {
+        Some(output) => {
+            if output.stdout.is_empty() {
+                "None".to_string()
+            } else {
+                String::from_utf8(output.stdout).unwrap_or("None".to_string())
+            }
+        }
+        None => "None".to_string(),
+    };
+    (c_sdk_hash, c_sdk_version)
+}
+
+fn retrieve_makefile_infos(bolos_sdk: &Path) -> Result<(u32, String), SDKBuildError> {
+    let makefile_defines =
+        File::open(bolos_sdk.join("Makefile.defines")).expect("Could not find Makefile.defines");
+    let mut api_level: i32 = -1;
+    let mut sdk_name = String::new();
+    for line in BufReader::new(makefile_defines).lines().flatten() {
+        if line.contains("API_LEVEL") {
+            api_level = line.split(":=").collect::<Vec<&str>>()[1]
+                .trim()
+                .parse()
+                .map_err(|_| SDKBuildError::InvalidAPILevel)?;
+        } else if line.contains("SDK_NAME") {
+            sdk_name = line.split(":=").collect::<Vec<&str>>()[1]
+                .trim()
+                .to_string();
+        }
+    }
+
+    if sdk_name.is_empty() || api_level == -1 {
+        return Err(SDKBuildError::CouldNotGetMakefileInfos);
+    } else {
+        return Ok((api_level as u32, sdk_name));
+    }
+}
+
+fn retrieve_target_file_infos(
+    device: &Device,
+    bolos_sdk: &Path,
+) -> Result<(String, String), SDKBuildError> {
+    let prefix = if *device == Device::NanoS {
+        "".to_string()
+    } else {
+        format!("target/{}/", device.to_string())
+    };
+    let target_file_path = bolos_sdk.join(format!("{}include/bolos_target.h", prefix));
+    println!("cargo:warning=Target file path is {:?}", target_file_path);
+    let target_file =
+        File::open(target_file_path).map_err(|_| SDKBuildError::TargetFileNotFound)?;
+    let mut target_id = String::new();
+    let mut target_name = String::new();
+    for line in BufReader::new(target_file).lines().flatten() {
+        if line.contains("TARGET_ID") {
+            let id = line.split(" ").collect::<Vec<&str>>()[2];
```

Comment:
`let id = line.split(" ").nth(2).unwrap();`

Reply:
Updated the way we iterate to make sure we only parse the values once.